src/main/diagnostics/abstractCodeActionProvider.ts

```diff
@@ -89,14 +89,34 @@ export abstract class AbstractCodeActionProvider implements vscode.CodeActionPro
             let diagnostic: vscode.Diagnostic = new vscode.Diagnostic(
                 new vscode.Range(dependencyPos[i], dependencyPos[i + 1]),
                 node.topSeverity === Severity.Normal ? 'No issues found.' : 'Top issue severity: ' + SeverityUtils.getString(node.topSeverity),
-                DiagnosticsUtils.getDiagnosticSeverity(node.topSeverity)
+                vscode.DiagnosticSeverity.Hint
             );
             diagnostic.source = this.getSource();
             diagnostic.code = node.componentId;
             diagnostics.push(diagnostic);
         }
     }
 
+    /**
+     * Add a new severity icon on the gutter.
+     * @param textEditor Gutter's editor.
+     * @param severity Severity icon.
+     * @param dependencyPos The dependency position in the descriptor.
+     */
+    addGutter(textEditor: vscode.TextEditor, severity: Severity, dependencyPos: vscode.Position[]) {
+        if (textEditor) {
+            for (let i: number = 0; i < dependencyPos.length; i += 2) {
```

Comment:
Why is the loop needed? Isn't the dependencyPos always sized 2?

Reply:
No. Maven, for instance, has 6 positions of 3 rows (gav).